tests-js/fido2.test.js

```diff
@@ -0,0 +1,35 @@
+import startFido2 from '../src/fido2.js'
+import fetchMock from 'fetch-mock'
+import 'babel-polyfill'
+
+describe('Fido 2', () => {
+
+    test('general test', () => {
+        const form = document.createElement('form')
+        form.id = 'django-fido-form'
+        form.dataset.mode = 'authentication'
+        form.dataset.url = '/data'
+        form.client_data = {value: 'client data'}
+        form.credential_id = {value: 'credential_id'}
+        form.authenticator_data = {value: 'authenticator_data'}
+        form.signature = {value: 'signature'}
+
+        const submit_button = document.createElement('button')
+        submit_button.id = 'submit-button'
+
+        fetchMock.mock('/data',
+            //   JSON.parse('{"rpId":"mojeid.mstrouhal","challenge":{"0":46,"1":123,"2":84,"3":210,"4":138,"5":140,"6":211,"7":127,"8":216,"9":81,"10":194,"11":1,"12":141,"13":8,"14":101,"15":197,"16":232,"17":133,"18":194,"19":153,"20":176,"21":110,"22":155,"23":252,"24":228,"25":5,"26":210,"27":71,"28":73,"29":68,"30":182,"31":128},"allowCredentials":[{"type":"public-key","id":{"0":193,"1":23,"2":244,"3":42,"4":225,"5":62,"6":115,"7":231,"8":177,"9":191,"10":166,"11":165,"12":19,"13":103,"14":248,"15":13,"16":247,"17":107,"18":154,"19":8,"20":196,"21":131,"22":125,"23":124,"24":42,"25":111,"26":243,"27":180,"28":241,"29":63,"30":31,"31":29,"32":42,"33":136,"34":66,"35":44,"36":114,"37":6,"38":226,"39":75,"40":46,"41":34,"42":48,"43":219,"44":181,"45":62,"46":87,"47":146,"48":231,"49":76,"50":226,"51":31,"52":207,"53":74,"54":107,"55":164,"56":213,"57":80,"58":157,"59":113,"60":183,"61":228,"62":15,"63":193}}],"timeout":30000,"userVerification":"preferred"}')
+            JSON.parse('{"publicKey":{"rpId":"mojeid.mstrouhal","challenge":"wuhwhe==","allowCredentials":[{"type":"public-key","id":"wuheehe"}],"timeout":30000,"userVerification":"preferred"}}')
+        )
+
+        document.getElementsByTagName('body')[0].appendChild(form)
+        document.getElementsByTagName('body')[0].appendChild(submit_button)
+
+        navigator.credentials = {get: jest.fn(public_key => {
+            expect(public_key).toMatchSnapshot()
+        })}
+
+        startFido2()
+    })
+
+})
```

Comment:
Check for registration should be also added.

Reply:
Fixed